python/pyarrow/tests/test_array.py

```diff
@@ -2666,6 +2666,31 @@ def test_array_masked():
     assert arr.type == pa.int64()
 
 
+def test_binary_array_masked():
+    # ARROW-12431
+    masked_basic = pa.array([b'\x05'], type=pa.binary(1),
+                            mask=np.array([False]))
+    assert [b'\x05'] == masked_basic.to_pylist()
+
+    # Fixed Length Binary
+    masked = pa.array(np.array([b'\x05']), type=pa.binary(1),
+                      mask=np.array([False]))
+    assert [b'\x05'] == masked.to_pylist()
+
+    masked_nulls = pa.array(np.array([b'\x05']), type=pa.binary(1),
+                            mask=np.array([True]))
+    assert [None] == masked_nulls.to_pylist()
+
+    # Variable Length Binary
+    masked = pa.array(np.array([b'\x05']), type=pa.binary(-1),
```

Comment:
This should be `pa.binary()`, not `pa.binary(-1)`.

Reply:
I thought that it was better to make it explicit that we were testing against a variable length binary, but given that ``-1``  is the default anyway I guess it doesn't make much difference.

Reply:
Well, `pa.binary()` is the official way to get a variable length binary. `binary(-1)` is an implementation detail.